azure-pipelines.yml

```diff
@@ -6,191 +6,202 @@ schedules:
-      scriptType: ps
-      inlineScript: >
-        az webapp restart --name netsourceindex --slot staging --resource-group source.dot.net
-
-  - pwsh: |
-      Start-Sleep 60
-      $urls = @(
-        "https://netsourceindex-staging.azurewebsites.net",
-        "https://netsourceindex-staging.azurewebsites.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/String.cs.html"
-      )
-      foreach ($url in $urls) {
-        $statusCode = Invoke-WebRequest $url -UseBasicParsing -SkipHttpErrorCheck | select -ExpandProperty StatusCode
-        if ($statusCode -ne 200) {
-          Write-Host "##vso[task.logissue type=error;]Deployed website returned undexpected status code $statusCode from url $url"
-          Write-Host "##vso[task.complete result=Failed;]Deployed website returned undexpected status code $statusCode from url $url"
-        }
-      }
-    displayName: Test Deployed WebApp
-
-  - task: AzureCLI@2
-    displayName: Swap Staging Slot into Production
-    inputs:
-      azureSubscription: SourceDotNet-Deployment-ARM
-      scriptLocation: inlineScript
-      scriptType: ps
-      inlineScript:  >
-        az webapp deployment slot swap
-        --resource-group source.dot.net
-        --name netsourceindex
-        --slot staging
-        --target-slot production
-
-  - task: AzureCLI@2
-    displayName: Cleanup Old Storage Containers
-    inputs:
-      azureSubscription: SourceDotNet-Deployment-ARM
-      scriptLocation: inlineScript
-      scriptType: ps
-      inlineScript:  >
-        deployment/cleanup-old-containers.ps1
-        -ResourceGroup source.dot.net
-        -WebappName netsourceindex
-        -StorageAccountName netsourceindex
-
-  - publish: $(Build.ArtifactStagingDirectory)/logs
-    artifact: logs
-    condition: always()
-
-  - publish: bin/repo
-    artifact: repo-logs
-    condition: always()
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: NetCore1ESPool-Internal-XL
+      image: 1es-windows-2022
+      os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+    - stage: stage
+      jobs:
+      - job: BuildIndex
+        displayName: Build Source Index
+        timeoutInMinutes: 360
+
+        variables:
+        - group: source-dot-net stage1 variables
+
+        templateContext:
+          outputs:
+          - output: nuget
+            displayName: 'NuGet push'
+            packageParentPath: '$(Build.ArtifactStagingDirectory)'
+            packagesToPush: '$(Build.ArtifactStagingDirectory)/packages/*.nupkg'
+            nuGetFeedType: 'internal'
+            publishVstsFeed: '9ee6d478-d288-47f7-aacc-f6e6d082ae6d/d1622942-d16f-48e5-bc83-96f4539e7601'
+          - output: pipelineArtifact
+            condition: always()
```

Comment:
is this required to publish even on failure?

Reply:
yes, `condition: always()` is how you make a step run regardless of earlier steps failing, and is commonly used for log upload.